# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../atop/segfault.patch
# Copyright (C) 2007 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- atop-1.21/photoproc.c.vanillag	2007-10-18 13:12:11.000000000 +0200
+++ atop-1.21/photoproc.c	2007-10-18 15:11:47.000000000 +0200
@@ -304,16 +304,19 @@
 	if ( (fp = fopen("stat", "r")) == NULL)
 		return 0;
 
-	(void) fgets(line, sizeof line, fp);
+	if (fgets(line, sizeof line, fp) == NULL) {
+		fclose (fp);
+		return 0;
+	}
 
 	sscanf(line, "%d", &pid);		/* fetch pid */
 
 	cmdhead = strchr (line, '(');		/* fetch commandname */
 	cmdtail = strrchr(line, ')');
-	if ( (nr = cmdtail-cmdhead-1) > sizeof command)
-		nr = sizeof command;
+	if ( (nr = cmdtail-cmdhead-1) >= sizeof command)
+		nr = sizeof command - 1;
 	memcpy(command, cmdhead+1, nr);
-	memset(&command[nr], 0, sizeof command - nr);
+	command[nr] = 0;
 
 	rtprio = policy = 0;
 
